### PR TITLE
Improve Code Coverage System.Data & System.Data.Common

### PR DESCRIPTION
**Pull request by Smartis2812**
Hello there,

My attempt to improve the code coverage.
Please give me feedback if i'm heading the right direction.
I'll tried to improve two existing test classes and create a new one.

Thanks.

### CODE REVIEW

File: src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs [hunk 1]
```diff
@@ -8,22 +8,70 @@ namespace System.Data.Common.Tests
 {
     public class DbExceptionTests
     {
+        [Fact]
+        public void DefaultArgumentsShowed()
```
Comment (stephentoub) [inline src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs]: Nit: I'm not sure what "Showed" means here.  Maybe "ParameterlessConstructor_UsesDefaults", or something like that?

File: src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs [hunk 2]
```diff
@@ -8,22 +8,70 @@ namespace System.Data.Common.Tests
...
+        {
+            var defaultException = new CustomDbException();
+            Assert.Equal("External component has thrown an exception.", defaultException.Message);
```
Comment (stephentoub) [inline src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs]: We generally don't validate error message likes this, as they may get localized, in which case the test will fail when run on non-English cultures.  You can instead just do a NonNull test, or assert that string.IsNullOrWhitespace is false, or something like that.

File: src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs [hunk 3]
```diff
@@ -8,22 +8,70 @@ namespace System.Data.Common.Tests
...
+            si.AddValue("HResult", 1);
+            si.AddValue("Source", string.Empty);
+
+            var sc = new System.Runtime.Serialization.StreamingContext();
+
+            var exFromSerializationInfo = new CustomDbException(si, sc);
+
+            Assert.Equal(string.Empty, exFromSerializationInfo.Message);
+            Assert.IsType<ArgumentException>(exFromSerializationInfo.InnerException);
+            Assert.Equal(1, exFromSerializationInfo.ErrorCode);
```
[the remark this answers is not in the document]
Comment (Smartis2812) [reply src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs]: @stephentoub Thanks for the feedback.
I'll rework it, as soon as you get a answer to this one.

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs]: Same as every other type:
1. Add an instance of the type to https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs, preferably using the ctor with the most overloads / or create multiple instances if you think coverage isn't sufficient.
2. Add an equality check as an extension method in https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs

Make sure to follow the pattern that we used for other types. To generate the serialization blobs I recommend to add a temporary test method and print out the base64 blobs for both netfx and netcoreapp. Use the BinaryFormatterHelpers.ToBase64String() method.

[the remark this answers is not in the document]
Comment (Smartis2812) [reply src/System.Data.Common/tests/System/Data/Common/DbExceptionTests.cs]: Sorry, I'm a little bit unsure about here.
Should i rework the test `ArgumentsFromSerializationInfo()`, or create another one?

**Pull request merged**